extensions/wikia/DesignSystem/DesignSystemHelper.class.php

```diff
@@ -50,13 +62,68 @@ public static function getSvg( $name, $classNames = '', $alt = '' ) {
 	 * @return SimpleXMLElement
 	 */
 	private static function getCachedSvg( $name ) {
-		if ( isset( self::$svgCache[ $name ] ) ) {
-			$xml = self::$svgCache[ $name ];
+		if ( isset( self::$svgCache[$name] ) ) {
+			$xml = self::$svgCache[$name];
 		} else {
 			$xml = simplexml_load_file( self::SVG_DIR . '/' . $name . '.svg' );
-			self::$svgCache[ $name ] = $xml;
+			self::$svgCache[$name] = $xml;
 		}
 
 		return $xml;
 	}
+
+	/**
+	 * @desc Renders text based on value of the `type` field, supports following types:
+	 *       - text
+	 *       - translatable-text
+	 *       - link-text
+	 *
+	 * @param array $fields
+	 *
+	 * @return string
+	 */
+	public static function renderText( $fields ) {
+		if ( $fields['type'] === 'text' ) {
+			return htmlspecialchars( $fields['value'] );
+		} elseif ( $fields['type'] === 'translatable-text' ) {
+			if ( isset( $fields['params'] ) ) {
+				$paramsRendered = [ ];
+
+				if ( !array_key_exists( $fields['key'], self::MESSAGE_PARAMS_ORDER ) ) {
+					WikiaLogger::instance()->error(
+						'Design System tried to render a message with params that we don\'t support, ignore params',
+						[
+							'messageKey' => $fields['key'],
+							'params' => $fields['params']
+						]
+					);
+				} else {
+					foreach ( self::MESSAGE_PARAMS_ORDER[$fields['key']] as $index => $paramKey ) {
+						$paramsRendered[] = self::renderText( $fields['params'][$paramKey] );
```

Comment:
What do you think about defining max level for recursion?
We shouldn't trust input data and in this case we can get into very deep recursion.